CMakeLists.txt

```diff
@@ -25,26 +20,49 @@ add_compile_options(
 	-Wstrict-prototypes
 )
 
-add_executable(csn
-	log.c
-	utils.c
-	chunk.c
-	chunker.c
-	caibx.c
-	sha.c
-	store.c
-	store-local.c
-	store-http.c
-	index.c
-	target.c
-	ui.c
-	main.c
-)
-target_link_libraries(csn ${ZSTD_LIBRARIES} ${CURL_LIBRARIES} ${CRYPTO_LIBRARIES})
-target_include_directories(csn PRIVATE ${ZSTD_INCLUDE_DIRS} ${CURL_INCLUDE_DIRS} ${CRYPTO_INCLUDE_DIRS})
+include(CheckIncludeFile)
 
-install(TARGETS csn RUNTIME DESTINATION bin)
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink csn \$ENV{DESTDIR}/\${CMAKE_INSTALL_PREFIX}/bin/casync)")
+CHECK_INCLUDE_FILE("linux/fs.h" LINUX_FS_FOUND)
+CHECK_INCLUDE_FILE("sys/disk.h" SYS_DISK_FOUND)
+
+if(${LINUX_FS_FOUND})
+	add_compile_definitions(USE_LINUX_FS)
+elseif(${SYS_DISK_FOUND})
+	add_compile_definitions(USE_SYS_DISK)
+endif()
+
+if(${BUILD_CSN})
```

Comment:
`${}` is not necessary here and to be honest I've not seen this in other CMake files. Is this a new style recommendation I've missed?